### ISSUE TITLE
Problems in undo after a change in deleteContents

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5-engine#1167]
Recently we merged a change in `deleteContents` algorithm: https://github.com/ckeditor/ckeditor5-engine/pull/1163. The changed looked fine, tests were okay and it was fixing a bug in collaborative editing. Unfortunately, it looks like it brought some regressions in undo: https://github.com/ckeditor/ckeditor5-undo/issues/65#issuecomment-336810386

There were a lot of bugs reported in undo in the last few days, but at least the linked one looks like being caused by using merge instead of delete to remove an empty element. It looks like the one letter "sticks" to the move operation from the merge, even though it should not.

**Comment by Reinmar** [comment]
I'm not sure I fully understand – we moved from "remove" to "merge", but both ways create correct models (I mean – `deleteContents()` works itself). So, the problem here is with transformations, yes?

**Comment by scofalik** [comment]
Exactly. Merge contains "sticky move operation". In the situation where an empty node is "merged" the move operation moves 0 nodes. But when it is transformed back, some nodes may stick to it. This should be turned off in undo - and as I checked `context.forceNotSticky` flag is set to `true` but it still didn't work.

**Comment by scofalik** [comment]
Funny enough, sticking to the "move 0 nodes sticky operation" is exactly what we wanted to achieve but for normal (not undo) transformation.
